src/pybind/mgr/orchestrator/module.py

```diff
@@ -684,43 +690,31 @@ def _daemon_add_misc(self, daemon_type=None, placement=None, inbuf=None):
         'name=fs_name,type=CephString '
         'name=placement,type=CephString,req=false',
         'Start MDS daemon(s)')
-    def _mds_add(self, fs_name, placement=None):
+    def _mds_add(self, fs_name, placement=None, inbuf=None):
         spec = ServiceSpec(
             'mds', fs_name,
             placement=PlacementSpec.from_string(placement),
         )
+
         completion = self.add_mds(spec)
         self._orchestrator_wait([completion])
         raise_if_exception(completion)
         return HandleCommandResult(stdout=completion.result_str())
 
     @_cli_write_command(
         'orch daemon add rgw',
-        'name=realm_name,type=CephString,req=false '
-        'name=zone_name,type=CephString,req=false '
+        'name=realm_name,type=CephString '
+        'name=zone_name,type=CephString '
```

Comment:
removing the req=falses means you must specify realm_name and zone_name. I guess if we're removing the ability to use -i <input json> then this is ok, but otherwise seems wierd.

However, if we are getting rid of the -i json then why do we pass in inbuf. If we never use it?

Reply:
> removing the req=falses means you must specify realm_name and zone_name. I guess if we're removing the ability to use -i then this is ok, but otherwise seems wierd.
> 
yep, but `realm_name` and `zone_name` are required either way...

allowing for only a ServiceSpec ---or-- params on the command line simplifies the usage/implementation and avoids setting precedence if a value is supplied on both the commandline and in the ServiceSpec.

> However, if we are getting rid of the -i json then why do we pass in inbuf. If we never use it?

To avoid this traceback:
```
$ ceph orch daemon add mds x -i mds.yaml
Error EINVAL: Traceback (most recent call last):
  File "/root/src/github.com/mgfritch/ceph/src/pybind/mgr/mgr_module.py", line 1157, in _handle_command
    return self.handle_command(inbuf, cmd)
  File "/root/src/github.com/mgfritch/ceph/src/pybind/mgr/orchestrator/_interface.py", line 110, in handle_command
    return dispatch[cmd['prefix']].call(self, cmd, inbuf)
  File "/root/src/github.com/mgfritch/ceph/src/pybind/mgr/mgr_module.py", line 308, in call
    return self.func(mgr, **kwargs)
  File "/root/src/github.com/mgfritch/ceph/src/pybind/mgr/orchestrator/_interface.py", line 72, in <lambda>
    wrapper_copy = lambda *l_args, **l_kwargs: wrapper(*l_args, **l_kwargs)
  File "/root/src/github.com/mgfritch/ceph/src/pybind/mgr/orchestrator/_interface.py", line 63, in wrapper
    return func(*args, **kwargs)
TypeError: _mds_add() got an unexpected keyword argument 'inbuf
```

`inbuf` is always provided when there are additional args on the cli 

... and in the case of `apply osd` it happens to be a DriveSpec